Sources/Foundation/JSONSerialization.swift

```diff
@@ -274,27 +338,29 @@ internal extension JSONSerialization {
         return .utf8
     }
     
-    static func parseBOM(_ bytes: UnsafePointer<UInt8>, length: Int) -> (encoding: String.Encoding, skipLength: Int)? {
-        if length >= 2 {
-            switch (bytes[0], bytes[1]) {
-            case (0xEF, 0xBB):
-                if length >= 3 && bytes[2] == 0xBF {
-                    return (.utf8, 3)
-                }
-            case (0x00, 0x00):
-                if length >= 4 && bytes[2] == 0xFE && bytes[3] == 0xFF {
-                    return (.utf32BigEndian, 4)
-                }
-            case (0xFF, 0xFE):
-                if length >= 4 && bytes[2] == 0 && bytes[3] == 0 {
-                    return (.utf32LittleEndian, 4)
-                }
-                return (.utf16LittleEndian, 2)
-            case (0xFE, 0xFF):
-                return (.utf16BigEndian, 2)
-            default:
-                break
+    static func parseBOM(_ bytes: UnsafeRawBufferPointer) -> (encoding: String.Encoding, skipLength: Int)? {
+        guard bytes.count >= 2 else {
```

Comment:
shouldn't we just write this as

```swift
if bytes.starts(with: [0xEF, 0xBB, 0xBF]) {
    return (.utf8, 3)
} else if bytes.starts(with: [0, 0]) {
    return (.utf32BigEndian, 4)
} else if ...
```

Reply:
sadly, the optimiser doesn't get that yet... Would need to pull the arrays into global/static `let utf8BOM = [0xEF, 0xBB, 0xBF]`.

Reply:
filed as https://bugs.swift.org/browse/SR-14102